packages/hoppscotch-common/src/components/share/CustomizeModal.vue

```diff
@@ -0,0 +1,454 @@
+  },
+  {
+    value: "button",
+    label: t("shared_requests.button"),
+  },
+  {
+    value: "link",
+    label: t("shared_requests.link"),
+  },
+]
+
+const parsedRequest = computed(() => {
+  if (!props.request) return null
+  return JSON.parse(props.request.request)
+})
+
+type EmbedTabs = "parameters" | "body" | "headers" | "authorization"
+
+type EmbedOption = {
+  selectedTab: EmbedTabs
+  tabs: {
+    value: EmbedTabs
+    label: string
+    enabled: boolean
+  }[]
+  theme: "light" | "dark" | "system"
+}
+
+const embedOptions = ref<EmbedOption>({
+  selectedTab: "parameters",
+  tabs: [
+    {
+      value: "parameters",
+      label: t("tab.parameters"),
+      enabled: true,
+    },
+    {
+      value: "body",
+      label: t("tab.body"),
+      enabled: true,
+    },
+    {
+      value: "headers",
+      label: t("tab.headers"),
+      enabled: true,
+    },
+    {
+      value: "authorization",
+      label: t("tab.authorization"),
+      enabled: true,
+    },
+  ],
+  theme: "system",
+})
+
+const embedThemeIcon = computed(() => {
+  if (embedOptions.value.theme === "system") {
+    return IconMonitor
+  } else if (embedOptions.value.theme === "light") {
+    return IconSun
+  } else {
+    return IconMoon
+  }
+})
+
+const removeEmbedOption = (option: EmbedTabs) => {
+  const index = embedOptions.value.tabs.findIndex((tab) => tab.value === option)
+  if (index === -1) return
+
+  //if removed tab is the selected tab, select the next tab with enabled true
+  if (embedOptions.value.selectedTab === option) {
+    const nextTab = embedOptions.value.tabs.find((tab) => tab.enabled)
+    if (nextTab) {
+      embedOptions.value.selectedTab = nextTab.value
+    }
+  }
+
+  embedOptions.value.tabs[index].enabled =
+    !embedOptions.value.tabs[index].enabled
+}
+
+type ButtonVariant = {
+  id: string
+  img: string
+}
+const buttonVariants: ButtonVariant[] = [
+  {
+    id: "button1",
+    img: "badge.svg",
+  },
+  {
+    id: "button2",
+    img: "badge-light.svg",
+  },
+  {
+    id: "button3",
+    img: "badge-dark.svg",
+  },
+]
+
+type LinkVariant = {
+  id: string
+  link?: string
+  label?: string
+  type: "html" | "markdown" | "link"
+}
+
+const linkVariants: LinkVariant[] = [
+  {
+    id: "link1",
+    link: props.request?.id,
+    type: "link",
+  },
+  {
+    id: "link2",
+    label: "shared_requests.run_in_hoppscotch",
+    type: "html",
+  },
+  {
+    id: "link3",
+    label: "shared_requests.run_in_hoppscotch",
+    type: "markdown",
+  },
+]
+
+const baseURL = import.meta.env.VITE_SHORTCODE_BASE_URL ?? "https://hopp.sh"
+
+const copyEmbed = () => {
+  const options = embedOptions.value
+  const enabledEmbedOptions = options.tabs
+    .filter((tab) => tab.enabled)
+    .map((tab) => tab.value)
+    .toString()
+  return `<iframe src="${baseURL}/e/${props.request?.id}/${enabledEmbedOptions}' style='width: 100%; height: 500px; border: 0; border-radius: 4px; overflow: hidden;'></iframe>`
```

Comment:
```suggestion
  return `<iframe src="${baseURL}/r/${props.request?.id}/${enabledEmbedOptions}" style='width: 100%; height: 500px; border: 0; border-radius: 4px; overflow: hidden;"></iframe>`
```

Currently, the embed gives a `404`. I think the route here should be `/r` with the embed options supplied via query params. Also, shouldn't tabs in the embed displayed be based on the above options?

Reply:
`/r/*` is reserved for shortcodes and it's redirected to the home page with it's resolved request setup.

We're hoping to reserve `/e/*` for embeds to render the embedded page.

Reply:
Gotcha, shouldn't there be a new page added under the `e` route?

Reply:
Yep, but that's part of another ticket. Out of scope of this feature PR.

Reply:
But can have a placeholder blank page in `/e` I guess. @nivedin can you add a blank page in `/e` path along with this PR?

Reply:
Or maybe mention it in the PR description such that the embed view implementation will be handled separately with a link to the corresponding Linear ticket.